Lib/ufo2ft/featureWriters/baseFeatureWriter.py

```diff
@@ -306,3 +307,49 @@ def compileGSUB(self):
         if compiler and not hasattr(compiler, "_gsub"):
             compiler._gsub = gsub
         return gsub
+
+    def getOpenTypeCategories(self):
+        """Return 'public.openTypeCategories' values as a tuple of lists of bases,
+        ligatures, marks, components."""
+        font = self.context.font
+        bases, ligatures, marks, components = set(), set(), set(), set()
+        openTypeCategories = font.lib.get(OPENTYPE_CATEGORIES_KEY, {})
+
+        for glyphName, category in openTypeCategories.items():
+            if category is None or category == "unassigned":
```

Comment:
When would `category` be `None`?

Reply:
Also, should an unrecognized category emit a warning?

Reply:
Yeah, i don't think it's possible for an individual category to be None in the parsed lib plist. And yes, it's good idea to emit a logging.warning with an unrecognized category (a hard crash would be too harsh)

Reply:
Oopsie, this was a leftover from a previous loop where is may have been None. Fixed now.

Reply:
Cool, please also add a warning for the `else:` arm.

Reply:
Done, with a test.